Dockerfile

```diff
@@ -67,6 +72,7 @@ RUN apt-get update && apt-get install -y \
 	xfsprogs \
 	libzfs-dev \
 	tar \
+	linux-image-`uname -r` \
```

Comment:
@estesp I don't want to make this a blocking issue, but I would like clarification regarding your statement:

> This Dockerfile starts with FROM ubuntu:14.04 and so I think that the expectation is the RUN command installing pre-reqs is strongly tied to that distro's package set, both in name and content.

I thought one of the purposes of Docker is to insulate the application running inside a container from the host system that it is running on.  But it seems like you are saying that to build Docker the user should be using a Ubuntu system because the Docker file says `FROM ubuntu:14.04`.   Doesn't that defeat some of the purpose of encapsulating the Docker build process within a container?  That means anyone using a non-Ubuntu host has to modify the Dockerfile in order to build Docker.  Not just that they need to follow different steps to install CRIU and other dependencies (something I think is reasonable).  But that they need to modify the Dockerfile source in order to build the docker static binary using a non-Ubuntu host.  So basically anyone wanting to build the experimental version of Docker on a non-Ubuntu host would need to fork the code, even if they had no interest in experimenting with the checkpoint feature.

Again I don't want this to be a blocking issue since I'm very interested in using this feature and I don't mind having mine own fork to make it work for me on Arch Linux.  But I wanted to comment since it seems like either my understanding of how the Docker build process is supposed to work is wrong, or this change might be more problematic for non-Ubuntu Linux users than initially imagined.